audio/filter/af_scaletempo2_internals.c

```diff
@@ -847,28 +819,27 @@ void mp_scaletempo2_init(struct mp_scaletempo2 *p, int channels, int rate)
     //                   1,          ...         |num_candidate_blocks|
     p->search_block_center_offset = p->num_candidate_blocks / 2
         + (p->ola_window_size / 2 - 1);
-    p->ola_window = realloc(p->ola_window, sizeof(float) * p->ola_window_size);
+    p->ola_window = talloc_realloc(p, p->ola_window, float, p->ola_window_size);
     get_symmetric_hanning_window(p->ola_window_size, p->ola_window);
-    p->transition_window = realloc(p->transition_window,
-        sizeof(float) * p->ola_window_size * 2);
+    p->transition_window = talloc_realloc(p, p->transition_window,
+        float, p->ola_window_size * 2);
     get_symmetric_hanning_window(2 * p->ola_window_size, p->transition_window);
 
     p->wsola_output_size = p->ola_window_size + p->ola_hop_size;
-    p->wsola_output = realloc_2d(p->wsola_output, p->channels, p->wsola_output_size);
-    // Initialize for overlap-and-add of the first block.
-    zero_2d(p->wsola_output, p->channels, p->wsola_output_size);
```

Comment:
Note: two zero_2d calls weren't needed before either, because overlap wasn't done for the first iteration any more.

Reply:
I would make such changes in separate commits to make it clear why it is removed.

Reply:
good point. I added the commit before the talloc one, so it's clear this was the case before the migration.